xbmc/network/GUIDialogNetworkSetup.h

```diff
@@ -25,17 +25,19 @@
 class CGUIDialogNetworkSetup : public CGUIDialogSettingsManualBase
 {
 public:
-  enum NET_PROTOCOL { NET_PROTOCOL_SMB = 0,
-                      NET_PROTOCOL_XBMSP,
-                      NET_PROTOCOL_FTP,
-                      NET_PROTOCOL_HTTP,
-                      NET_PROTOCOL_HTTPS,
-                      NET_PROTOCOL_DAV,
-                      NET_PROTOCOL_DAVS,
-                      NET_PROTOCOL_UPNP,
-                      NET_PROTOCOL_RSS,
-                      NET_PROTOCOL_SFTP,
-                      NET_PROTOCOL_NFS};
+  //! \brief A structure encapsulating properties of a supported protocol.
+  struct Protocol
+  {
+    bool supportPath;      //!< Protocol has path in addition to server name
+    bool supportUsername;  //!< Protocol uses logins
+    bool supportPassword;  //!< Protocol supports passwords
+    bool supportPort;      //!< Protocol supports port customization
+    bool supportBrowsing;  //!< Protocol supports server browsing
+    int defaultPort;       //!< Default port to use for protocol
+    std::string type;      //!< URL type for protocol
+    int label;             //!< String ID to use as label in dialog
```

Comment:
This could be reordered to avoid some padding. Not really something we've cared about but it's been in the back of my mind recently to look into

Reply:
Why would we care about padding for this struct?